src/core/models.py

```diff
@@ -1906,3 +1923,558 @@ def log_hijack_ended(sender, hijacker, hijacked, request, **kwargs):
+    )
+    ror_status = models.CharField(
+        blank=True,
+        max_length=10,
+        choices=RORStatus.choices,
+        default=RORStatus.UNKNOWN,
+    )
+    website = models.CharField(
+        blank=True,
+        max_length=500,
+    )
+    locations = models.ManyToManyField(
+        'Location',
+        blank=True,
+        null=True,
+    )
+
+    class Meta:
+        ordering = ['ror_display__value']
+        constraints = [
+            models.UniqueConstraint(
+                fields=['ror'],
+                condition=~models.Q(ror__exact=''),
+                name='filled_unique',
+            )
+        ]
+
+    def __str__(self):
+        elements = [
+            str(self.name) if self.name else '',
+            str(self.location) if self.location else '',
+        ]
+        return ', '.join([element for element in elements if element])
+
+    @property
+    def name(self):
+        """
+        Return the OrganizationName that ROR uses for display, or if none,
+        the first one that was manually entered, or if none
+        the first one designated by ROR as a label.
+        Can be expanded in future to support choosing a label by language.
+        """
+        try:
+            return self.ror_display
+        except Organization.ror_display.RelatedObjectDoesNotExist:
+            try:
+                return self.custom_label
+            except Organization.custom_label.RelatedObjectDoesNotExist:
+                return self.labels.first()
+
+    @property
+    def location(self):
+        """
+        Return the first location.
+        """
+        return self.locations.first() if self.locations else None
+
+    @property
+    def names(self):
+        """
+        All names.
+        """
+        return OrganizationName.objects.filter(
+            models.Q(ror_display_for=self) |
+            models.Q(custom_label_for=self) |
+            models.Q(label_for=self) |
+            models.Q(alias_for=self) |
+            models.Q(acronym_for=self),
+        )
+
+    @property
+    def also_known_as(self):
+        """
+        All names excluding the ROR display name.
+        """
+        return self.names.exclude(ror_display_for=self)
+
+    @classmethod
+    def naive_get_or_create(
+        cls,
+        value,
+        account=None,
+        frozen_author=None,
+        preprint_author=None,
+    ):
+        """
+        Backwards-compatible API for finding a matching organization name.
+        Intended for use in batch importers where ROR data is not available
+        in the data being imported.
+        Does not support ROR ids, ROR name types, or location data.
+        """
+        # Is there a single exact match in the
+        # canonical name data from ROR (e.g. labels)?
+        try:
+            organization = cls.objects.get(labels__value=value)
+            created = False
+        except cls.DoesNotExist or cls.MultipleObjectsReturned:
```

Comment:
This wont catch `cls.MultipleObjectsReturned`. It should be a tuple `except (cls.DoesNotExist, cls.MultipleObjectsReturned):`